### PR TITLE
Rich text

### PR DESCRIPTION
**Pull request by oestradiol**

### CODE REVIEW

File: src/components/PostComposer.tsx [hunk 1]
```diff
@@ -30,15 +39,15 @@ const PostComposer: Component<{ setUnreadCount: Setter<number> }> = ({
...
+              ...recordData,
             } as SocialPskyFeedPost.Record,
           },
         })
         .catch((err) => console.log(err));
     } else {
       await fetch(`https://${SERVER_URL}/post`, {
         method: "POST",
-        body: JSON.stringify({ post: post }),
+        body: JSON.stringify(recordData),
```
Comment (oestradiol) [inline src/components/PostComposer.tsx:50]: i forgor it uses post, lemme fix that

Comment (oestradiol) [reply src/components/PostComposer.tsx:50]: actually we should just update the backend tbh

File: src/utils/rich-text/detection.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    if (!uri.startsWith("http")) {
+      const domain = match.groups?.domain;
+      if (!domain || !isValidDomain(domain)) {
+        continue;
+      }
+      uri = `https://${uri}`;
+    }
+    const start = text.utf16.indexOf(match[2], match.index);
+    const index = { start, end: start + match[2].length };
+    // strip ending puncuation
```
Comment (oestradiol) [inline src/utils/rich-text/detection.ts:60]: LOL the typo came with it skull

Comment (oestradiol) [reply src/utils/rich-text/detection.ts:60]: i'll keep it there just because it's funny

File: src/components/PostComposer.tsx (hunk 1 above)
Comment (notjuliet) [reply src/components/PostComposer.tsx]: this part of the code will be removed eventually, i dont see myself supporting anonposting on the long term, eventually oauth will be required

Comment (oestradiol) [reply src/components/PostComposer.tsx]: i mean, fair but until we do so we should be supporting every feature
also, i'll have to update the backend regardless to update the lexicon so i might as well

**Comment by oestradiol** [comment]
Backend updates: https://github.com/notjuliet/picosky/pull/2

**Review by notjuliet — APPROVED** [review]

**Pull request merged**